components/feature_restricted_modal/feature_restricted_modal.tsx

```diff
@@ -20,64 +20,65 @@ import {isModalOpen} from 'selectors/views/modals';
 import {GlobalState} from 'types/store';
 import {ModalIdentifiers, AboutLinks, LicenseLinks} from 'utils/constants';
 
-import './create_team_restricted_modal.scss';
+import './feature_restricted_modal.scss';
 
-const CreateTeamRestrictedModal = () => {
+type FeatureRestrictedModalProps = {
+    modalTitle: string;
+    modalMessage: string;
+    modalTitleAfterTrial: string;
+    modalMessageAfterTrial: string;
+}
+
+const FeatureRestrictedModal = ({
+    modalTitle,
+    modalMessage,
+    modalTitleAfterTrial,
+    modalMessageAfterTrial,
+}: FeatureRestrictedModalProps) => {
     const {formatMessage} = useIntl();
     const dispatch = useDispatch<DispatchFunc>();
 
     const subscription = useSelector(selectCloudSubscription);
     const isSystemAdmin = useSelector(isCurrentUserSystemAdmin);
-    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.CREATE_TEAM_RESTRICTED_MODAL));
+    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.FEATURE_RESTRICTED_MODAL));
     const openPricingModal = useOpenPricingModal();
 
     if (!show) {
         return null;
     }
 
-    const hadPrevCloudTrial = subscription?.is_free_trial === 'false' && subscription?.trial_end_at > 0;
-    const showStartTrial = isSystemAdmin && !hadPrevCloudTrial;
+    const dismissAction = () => {
+        dispatch(closeModal(ModalIdentifiers.FEATURE_RESTRICTED_MODAL));
+    };
 
-    const dismissAction = async () => {
-        await dispatch(closeModal(ModalIdentifiers.CREATE_TEAM_RESTRICTED_MODAL));
+    const handleViewPlansClick = () => {
+        openPricingModal();
+        dismissAction();
     };
 
+    const hadPrevCloudTrial = subscription?.is_free_trial === 'false' && subscription?.trial_end_at > 0;
```

Comment:
nit: there is a selector in `selectors/entities/cloud.ts`called checkHadPriorTrial that does this for us.